src/store.js

```
@@ -11,6 +11,13 @@ export class OverriddenComponentRepository {
     this.components[id] = Component;
   };
 
+  append = (id, Component) => {
+    if (!Array.isArray(this.components[id])) {
```

Should this fail if there's already a single-component override? Or automatically convert the existing entry to an array containing that one?

Right now `add` followed by `append` silently discards the initial element.

---

I would say that is debatable. 

After thinking it through i would say, `append` should silently convert the existing one element to an array with the element in it. 

BUT: what do we do if afterwards a `add` is called?

after thinking that another time through (i already feel like i am a reasoning model) i would use another property for `append`

```
  append = (id, Component) => {
    if (!Array.isArray(this.components_multiple[id])) { // TODO: naming
      this.components_multiple[id] = [];
    }
    this.components_multiple[id].push(Component);
  };
  ```

with following change:

```
  get = id => {
    let out = [];
    if (id in this.components_multiple) {
       out = this.components_multiple[id];
    }

    if (id in this.components) {
      out.push(this.components[id])
    }

    if (out.length > 1) {
       return out;
    } else {
        return out[0];
    }
  };
```

with this approach we would enable the option to add another item to a container which is already replaced.